graftm/unpack_sequences.py

```diff
@@ -35,6 +43,32 @@ class UnexpectedFileFormatException(Exception): pass
     
     def __init__(self, read_file):
         self.read_file   = read_file
+        self.gapped_alignment = False
+        self.file_type = self._guess_sequence_input_file_format(self.read_file)
+        self.type = self._guess_sequence_type(self.read_file)
+        
+    def _ungap_sequences(self):
+        logging.debug("Removing gaps from the input sequences")
+        if self.format()==self.FORMAT_FASTA:
+            reads=SeqIO.parse(open(self.read_file), "fasta")
+        elif self.format()==self.FORMAT_FASTQ:
+            reads=SeqIO.parse(open(self.read_file), "fastq")
+        else:
+            logging.warning("If this is a large sequence file this is going to take some time - Did I mention inputting alignments to GraftM is probably a bad idea?")
```

Comment:
How about instead iterating through the file until 100 non-gap characters are encountered, instead of dumping the entire file?